src/tasks/copy-template-files.ts

```diff
@@ -213,7 +217,32 @@ const processTemplatedFiles = async (
         templateFileDescriptor.relativePath.split(templateTargetName)[0],
         templateTargetName,
       );
-      fs.writeFileSync(targetPath, output);
+
+      let finalOutput = output;
+
+      const containsNextjs = targetPath.includes("nextjs");
+      const containsDotfile = /\/\.[^/]+/.test(targetPath);
+      const hardhatOrFoundry = targetPath.split("packages")[1]?.split("/")[1];
+      const isHardhatOrFoundry =
+        hardhatOrFoundry && (hardhatOrFoundry.includes("hardhat") || hardhatOrFoundry.includes("foundry"));
+
+      if (containsNextjs && !containsDotfile) {
+        finalOutput = await formatWithPrettier(output, {
+          ...nextJsPrettierConfig,
+          parser: "typescript",
+          plugins: [pluginSorter],
+          fallbackConfig: { parser: "typescript", plugins: undefined },
+        });
+      } else if (isHardhatOrFoundry && !containsDotfile) {
+        finalOutput = await formatWithPrettier(output, {
+          ...hardhatPrettierConfig,
+          parser: "solidity-parse",
+          plugins: [solidityPlugin],
+          fallbackConfig: { parser: "typescript", plugins: undefined },
+        });
+      }
+
+      fs.writeFileSync(targetPath, finalOutput);
```

Comment:
Seeing here `hardhat` & `foundry` tells me that we might be coupling this too much. This is an already complex function, and contains the core/most important logic of the CLI.

So it makes me wonder if there is an alternative: couldn't we run the "yarn format" somehow in the instance as we were doing before... but using the create-eth prettier (instead of the instance one.. that it's not installed when they omit installing deps)

Reply:
> This is an already complex function, and contains the core/most important logic of the CLI.

Yeah agree, even I felt this while writing it. 

>  couldn't we run the "yarn format" somehow in the instance as we were doing before

Yeah actually this was first intuition but didn't find any good resources on how to achieve it and felt complex, but will be doing more research on this and also I feel this is more robust way to do it.